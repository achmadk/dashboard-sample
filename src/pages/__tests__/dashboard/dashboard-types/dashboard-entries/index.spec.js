import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import chai from 'chai'

import AverageTenderDurationAsyncContainer, {
  loadDashboardEntries
} from '@/pages/dashboard/dashboard-types/dashboard-entries'
import AverageTenderDurationContainer from '@/pages/dashboard/dashboard-types/dashboard-entries/dashboard-entries'
import ProviderStore from '@/../test/provider-store'

describe('Dashboard Entries Container', () => {
  it('render correctly', () => {
    const wrapper = shallow(<AverageTenderDurationAsyncContainer />)
    expect(toJson(wrapper, { mode: 'deep' })).toMatchSnapshot()
  })

  it('return Dashboard component successfully', () => {
    const result = loadDashboardEntries()
    chai.expect(result).to.exist
  })
})

describe('Dashboard Entries Component', () => {
  it('render correctly', () => {
    const wrapper = shallow(
      <ProviderStore>
        <AverageTenderDurationContainer />
      </ProviderStore>
    )
    expect(toJson(wrapper, { mode: 'deep' })).toMatchSnapshot()
  })
})
