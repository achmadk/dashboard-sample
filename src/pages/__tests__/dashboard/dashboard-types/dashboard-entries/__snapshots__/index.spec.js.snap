// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Entries Component render correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <ConnectFunction />
</Provider>
`;

exports[`Dashboard Entries Container render correctly 1`] = `
<React.Suspense
  fallback={
    <div>
      Loading...
    </div>
  }
>
  <React.Lazy />
</React.Suspense>
`;
